src/arc/mp/MultiplayerSingleton.as

```diff
@@ -511,7 +511,9 @@ package arc.mp
 
         public function gameplaySubmit(room:Object):void
         {
-            if (currentSong.engine != null)
+            var matchSong:Object = currentSong || room.match.song;
+
+            if (matchSong != null && matchSong.engine != null)
```

Comment:
There's a chance `matchSong` is `null` and cause the code that sets the game stats string to fail. Perhaps change the check to `if (matchSong == null || matchSong.engine != null)` instead?

Reply:
As mentionned in devchat, this is a valid concern, but for a future PR.